process/notifier/sovereignNotifier.go

```diff
@@ -1,33 +1,199 @@
+		return err
+	}
+
+	extendedHeader := &block.ShardHeaderExtended{
+		Header:             headerV2,
+		IncomingMiniBlocks: mbs,
+	}
+
+	notifier.notifyHandlers(extendedHeader)
+	return nil
+}
+
+func checkNilOutportBlockFields(outportBlock *outport.OutportBlock) error {
+	if outportBlock == nil {
+		return errNilOutportblock
+	}
+	if outportBlock.TransactionPool == nil {
+		return errNilTransactionPool
+	}
+	if outportBlock.BlockData == nil {
+		return errNilBlockData
+	}
+
+	return nil
+}
+
+func (notifier *sovereignNotifier) getAllIncomingMbs(txPool *outport.TransactionPool) ([]*block.MiniBlock, error) {
+	mbs := make([]*block.MiniBlock, 0)
+
+	txsMb, err := notifier.getIncomingMbFromTxs(txPool.Transactions)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: when specs are defined, we should also handle scrs mbs
+	mbs = append(mbs, txsMb)
+	return mbs, nil
+}
+
+func (notifier *sovereignNotifier) getIncomingMbFromTxs(txs map[string]*outport.TxInfo) (*block.MiniBlock, error) {
+	txHashes := make([][]byte, 0)
+	execOrderTxHashMap := make(map[string]uint32)
+
+	for txHash, tx := range txs {
+		receiver := tx.GetTransaction().GetRcvAddr()
+		_, found := notifier.subscribedAddresses[string(receiver)]
+		if !found {
+			continue
+		}
+
+		hashBytes, err := hex.DecodeString(txHash)
+		if err != nil {
+			return nil, fmt.Errorf("%w, hash: %s", err, txHash)
+		}
+
+		log.Info("found incoming tx", "tx hash", txHash, "address", hex.EncodeToString(receiver))
+
+		txHashes = append(txHashes, hashBytes)
+		execOrderTxHashMap[string(hashBytes)] = tx.GetExecutionOrder()
+	}
+
+	sort.SliceStable(txHashes, func(i, j int) bool {
+		return execOrderTxHashMap[string(txHashes[i])] < execOrderTxHashMap[string(txHashes[j])]
+	})
+
+	return &block.MiniBlock{
+		TxHashes:        txHashes,
+		ReceiverShardID: 0, // todo: decide what we should fill here
+		SenderShardID:   0, // todo: decide what we should fill here
+		Type:            block.TxBlock,
+		Reserved:        nil,
+	}, nil
+}
+
+func (notifier *sovereignNotifier) getHeaderV2(headerType core.HeaderType, headerBytes []byte) (*block.HeaderV2, error) {
+	if headerType != core.ShardHeaderV2 {
+		return nil, fmt.Errorf("%w : %s, expected: %s",
+			errReceivedHeaderType, headerType, core.ShardHeaderV2)
+	}
+
+	headerHandler, err := block.GetHeaderFromBytes(notifier.marshaller, notifier.headerV2Creator, headerBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return headerHandler.(*block.HeaderV2), nil
+}
+
+func (notifier *sovereignNotifier) notifyHandlers(extendedHeader *block.ShardHeaderExtended) {
+	notifier.mutHandler.RLock()
+	defer notifier.mutHandler.RUnlock()
+
+	for _, handler := range notifier.handlers {
+		handler.SaveExtendedHeader(extendedHeader)
+	}
+}
+
+// RegisterHandler will register an extended header handler to be notified about incoming headers and miniblocks
+func (notifier *sovereignNotifier) RegisterHandler(handler process.ExtendedHeaderHandler) error {
+	if handler == nil {
```

Comment:
```suggestion
	if check.IfNil(handler) {
```
+ `IsInterfaceNil` on `ExtendedHeaderHandler` interface

Reply:
Changed as suggested